terraform/priv-server-a.tf

```diff
@@ -0,0 +1,72 @@
+data "aws_ami" "ubuntu" {
+  most_recent = true
+
+  # from terraform docs example lookup
+  filter {
+    name   = "name"
+    values = ["ubuntu/images/hvm-ssd/ubuntu-focal-20.04-amd64-server-*"]
+  }
+
+  # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/virtualization_types.html
+  filter {
+    name   = "virtualization-type"
+    values = ["hvm"]
+  }
+
+  owners = ["099720109477"] # Canonical (ubuntu)
+}
+
+
+# firewall
+resource "aws_security_group" "allow_ssh" {
+  name        = "allow_ssh"
+  description = "Allow ssh inbound traffic"
+  vpc_id      = aws_vpc.main.id
+
+  # allow from public_a only
+  ingress {
+    description = "SSH from VPC public subnets"
+    from_port   = 22
+    to_port     = 22
+    protocol    = "tcp"
+    cidr_blocks = [aws_subnet.private_a.cidr_block]
```

Comment:
You are to use the vpc_cidr block instead of the subnet cidr block
change this to `cidr_blocks = [aws_vpc.main.cidr_block]`

Reply:
Done!